### PR TITLE
feat: camunda exporters support tenant add and remove entity events

### LINKED ISSUE #24590: Camunda exporters support tenant add entity events

### LINKED ISSUE #24591: Camunda exporters support tenant remove entity events

### PR DESCRIPTION
**Pull request by Oleksiivanov**
## Description

- added TenantEntityAddedHandler and tests
- added TenantEntityRemoveHandler and tests
- updated index, entity and related code to join relation

## Checklist

- [ ] for CI changes:
  - [ ] structural/foundational changes signed off by [CI DRI](https://github.com/cmur2)
  - [ ] [ci.yml](https://github.com/camunda/camunda/blob/main/.github/workflows/ci.yml) modifications comply with ["Unified CI" requirements](https://github.com/camunda/camunda/wiki/CI-&-Automation#workflow-inclusion-criteria)

## Related issues

closes #24590 and #24591

### CODE REVIEW

File: zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/handlers/TenantEntityAddedHandler.java [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    // the script to add a single value to the assignedMemberKeys field
+    final String script =
+        "if (ctx._source.assignedMemberKeys == null) "
+            + "{ ctx._source.assignedMemberKeys = [params.newKey]; } "
+            + "else if (!ctx._source.assignedMemberKeys.contains(params.newKey)) "
+            + "{ ctx._source.assignedMemberKeys.add(params.newKey); }";
+
+    final Map<String, Object> params =
+        Map.of("newKey", entity.getAssignedMemberKeys().iterator().next());
+
```
[the remark this answers is not in the document]
Comment (Ben-Sheppard) [reply zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/handlers/TenantEntityAddedHandler.java:74]: I'm going to tag the exporter team as they may have better opinions than myself :)

**Comment by koevskinikola** [comment]
@Oleksiivanov I can start with the PR review here if no one else has started working on it yet.

File: zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/handlers/TenantCreateUpdateHandler.java [hunk 2]
```diff
@@ -56,7 +56,10 @@ public TenantEntity createNewEntity(final String id) {
   @Override
   public void updateEntity(final Record<TenantRecordValue> record, final TenantEntity entity) {
     final TenantRecordValue value = record.getValue();
-    entity.setKey(value.getTenantKey()).setTenantId(value.getTenantId()).setName(value.getName());
+    entity
+        .setTenantKey(value.getTenantKey())
+        .setTenantId(value.getTenantId())
+        .setName(value.getName());
```
Comment (koevskinikola) [inline zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/handlers/TenantCreateUpdateHandler.java:62]: ❌ We need to add a "parent" join relation here, do define this as the "parent" entity:

```suggestion
    final var joinRelation = TenantIndex.JOIN_RELATION_FACTORY.createParent();
    entity
        .setTenantKey(value.getTenantKey())
        .setTenantId(value.getTenantId())
        .setName(value.getName())
        .setJoin(joinRelation);
```

**Review by koevskinikola — CHANGES_REQUESTED** [review]
Nice job @Oleksiivanov, just a small change in the create-update handler and we can merge it 🚀

**Review by koevskinikola — APPROVED** [review]

**Pull request merged**